File: app/src/main/java/com/legue/axel/lolsummonertool/utils/ImageUtils.java
package com.legue.axel.lolsummonertool.utils;

import android.net.Uri;

import com.legue.axel.lolsummonertool.network.retrofit.RetrofitConstants;

public class ImageUtils {

    public static Uri BuildChampionIconUrl(String endPointUrl) {
        return Uri.parse(RetrofitConstants.API_DRAGON_BASE_CDN + RetrofitConstants.API_CHAMPION_VERSION + "/" + RetrofitConstants.API_TYPE_IMAGE + "/" + RetrofitConstants.API_TYPE_CHAMPION + "/" + endPointUrl);
    }

    public static Uri BuildItemIconUrl(String endPointUrl) {
        return Uri.parse(RetrofitConstants.API_DRAGON_BASE_CDN + RetrofitConstants.API_CHAMPION_VERSION + "/" + RetrofitConstants.API_TYPE_IMAGE + "/" + RetrofitConstants.API_TYPE_ITEM + "/" + endPointUrl);
    }

    public static Uri BuildMasteryIconUrl(String endPointUrl) {
        return Uri.parse(RetrofitConstants.API_DRAGON_BASE_CDN + RetrofitConstants.API_MASTERY_VERSION + "/" + RetrofitConstants.API_TYPE_IMAGE + "/" + RetrofitConstants.API_TYPE_MASTERY + "/" + endPointUrl);
    }
    public static Uri BuildSummonerSpellIconUrl(String endPointUrl) {
        return Uri.parse(RetrofitConstants.API_DRAGON_BASE_CDN + RetrofitConstants.API_SUMMONER_VERSION + "/" + RetrofitConstants.API_TYPE_IMAGE + "/" + RetrofitConstants.API_TYPE_SPELL + "/" + endPointUrl);
    }

    public static Uri BuildSpellIconUrl(String endPointUrl) {
        return Uri.parse(RetrofitConstants.API_DRAGON_BASE_CDN + RetrofitConstants.API_SPELL_VERSION + "/" + RetrofitConstants.API_TYPE_IMAGE + "/" + RetrofitConstants.API_TYPE_SPELL + "/" + endPointUrl);
    }

    public static Uri BuildPassiveIconUrl(String endPointUrl) {
        return Uri.parse(RetrofitConstants.API_DRAGON_BASE_CDN + RetrofitConstants.API_SPELL_VERSION + "/" + RetrofitConstants.API_TYPE_IMAGE + "/" + RetrofitConstants.API_TYPE_PASSIVE + "/" + endPointUrl);
    }


}

File: app/src/main/java/com/legue/axel/lolsummonertool/utils/ImageUtils.kt
package com.legue.axel.lolsummonertool.utils

import android.net.Uri

import com.legue.axel.lolsummonertool.network.retrofit.RetrofitConstants

object ImageUtils {

    fun buildChampionIconUrl(endPointUrl: String): Uri {
        return Uri.parse(RetrofitConstants.API_DRAGON_BASE_CDN + RetrofitConstants.API_CHAMPION_VERSION + "/" + RetrofitConstants.API_TYPE_IMAGE + "/" + RetrofitConstants.API_TYPE_CHAMPION + "/" + endPointUrl)
    }

    fun buildItemIconUrl(endPointUrl: String): Uri {
        return Uri.parse(RetrofitConstants.API_DRAGON_BASE_CDN + RetrofitConstants.API_CHAMPION_VERSION + "/" + RetrofitConstants.API_TYPE_IMAGE + "/" + RetrofitConstants.API_TYPE_ITEM + "/" + endPointUrl)
    }

    fun buildMasteryIconUrl(endPointUrl: String): Uri {
        return Uri.parse(RetrofitConstants.API_DRAGON_BASE_CDN + RetrofitConstants.API_MASTERY_VERSION + "/" + RetrofitConstants.API_TYPE_IMAGE + "/" + RetrofitConstants.API_TYPE_MASTERY + "/" + endPointUrl)
    }

    fun buildSummonerSpellIconUrl(endPointUrl: String): Uri {
        return Uri.parse(RetrofitConstants.API_DRAGON_BASE_CDN + RetrofitConstants.API_SUMMONER_VERSION + "/" + RetrofitConstants.API_TYPE_IMAGE + "/" + RetrofitConstants.API_TYPE_SPELL + "/" + endPointUrl)
    }

    fun buildSpellIconUrl(endPointUrl: String): Uri {
        return Uri.parse(RetrofitConstants.API_DRAGON_BASE_CDN + RetrofitConstants.API_SPELL_VERSION + "/" + RetrofitConstants.API_TYPE_IMAGE + "/" + RetrofitConstants.API_TYPE_SPELL + "/" + endPointUrl)
    }

    fun buildPassiveIconUrl(endPointUrl: String): Uri {
        return Uri.parse(RetrofitConstants.API_DRAGON_BASE_CDN + RetrofitConstants.API_SPELL_VERSION + "/" + RetrofitConstants.API_TYPE_IMAGE + "/" + RetrofitConstants.API_TYPE_PASSIVE + "/" + endPointUrl)
    }


}
